README.md

```diff
@@ -5,140 +5,507 @@
+```
+
+## Coordination primitives
+
+Coordination primitives can be used to elect leaders and synchronize access to shared resources
+in the cluster. These primitives typically require the use of consensus but allow certain consistency
+models to be relaxed, e.g. on reads. Coordination primitives include:
+* [`DistributedLock`](#distributedlock) - coarse grained fair distributed lock
+* `LeaderElection` - single leader election
+* [`LeaderElector`](#leaderelector) - multiple leader election with automatic leader balancing
+* `WorkQueue` - distributed persistent work queue with ack/fail mechanisms
+
+### DistributedLock
+
+As with many other objects in Atomix, primitives are constructed using a builder pattern. All
+core primitive builders are exposed via the `Atomix` API:
+
+```java
+DistributedLock lock = atomix.lockBuilder("test-lock")
+  .withLockTimeout(Duration.ofSeconds(2))
+  .buld();
+```
+
+The _lock timeout_ is the approximate time it takes to determine whether a lock has failed.
+If a lock's underlying Raft session is expired, the lock will be released and granted to the
+next waiting process.
+
+```java
 lock.lock();
 try {
-  // Do stuff...
+  // Do something
 } finally {
   lock.unlock();
 }
 ```
 
-#### Asynchronous distributed lock
+As with all Atomix primitives, a fully asynchronous version of the lock can be constructed by
+simply calling the `async()` method on the lock:
+
+```java
+AsyncDistributedLock asyncLock = lock.async();
+lock.lock().thenRun(() -> {
+  // Do something
+  lock.unlock().thenRun(() -> {
+    // Unlocked
+  });
+});
+```
+
+### LeaderElector
+
+The `LeaderElector` primitive is used to elect multiple leaders within a cluster, automatically
+balancing leaders among all instances of the primitive across the cluster.
+
 ```java
-AsyncDistributedLock lock = atomix.lockBuilder()
-  .withName("my-lock")
-  .buildAsync();
+LeaderElector<NodeId> elector = atomix.leaderElectorBuilder("test-elector")
+  .withElectionTimeout(Duration.ofSeconds(2))
+  .build();
+```
 
-lock.lock().thenAccept(version -> {
-  // Do stuff...
-  lock.unlock();
+As with `DistributedLock`, the `LeaderElector` (and `LeaderElection`) primitive supports a timeout
+after which inactive leaders will be demoted and a new leader will be elected, e.g. in the case of
+a crash or network partition. Similarly, as with all primitives, an `AsyncLeaderElector` can be
+created via the `async()` getter:
+
+```java
+AsyncLeaderElector<NodeId> asyncElector = elector.async();
+```
+
+To enter into an election, use the `run` method:
+
+```java
+asyncElector.run("foo", atomix.getClusterService().getLocalNode().id()).thenAccept(leadership -> {
+  if (leadership.leader().id().equals(atomix.getClusterService().getLocalNode().id())) {
+    // Local node elected leader!
+  }
+});
+```
+
+The topic passed as the first argument to the `run` method is an identifier for the leader election
+to run into. The leader elector will balance the leaders for all topics.
```

Comment:
... "method is an identifier for the leader election to run into". This seems confusing to me, maybe:

..." an identifier for the topic that the leader election runs for leadership of." ?

Reply:
Uhh... yeah I don't think that's what I meant to type :-P